lib/ruby_lsp/ruby_lsp_rails/rails_test_style.rb

```diff
@@ -0,0 +1,104 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Rails
+    class RailsTestStyle < Listeners::TestDiscovery
+      extend T::Sig
+
+      #: (RunnerClient client, ResponseBuilders::TestCollection response_builder, GlobalState global_state, Prism::Dispatcher dispatcher, URI::Generic uri) -> void
+      def initialize(client, response_builder, global_state, dispatcher, uri)
+        super(response_builder, global_state, dispatcher, uri)
+
+        dispatcher.register(
+          self,
+          :on_class_node_enter,
+          :on_call_node_enter,
+          :on_def_node_enter,
+        )
+      end
+
+      #: (Prism::ClassNode node) -> void
+      def on_class_node_enter(node)
+        with_test_ancestor_tracking(node) do |name, ancestors|
+          if declarative_minitest?(ancestors, name)
+            test_item = Requests::Support::TestItem.new(
+              name,
+              name,
+              @uri,
+              range_from_node(node),
+              framework: :rails,
+            )
+
+            @response_builder.add(test_item)
+          end
+        end
+      end
+
+      #: (Prism::CallNode node) -> void
+      def on_call_node_enter(node)
+        return unless node.name == :test
+        return unless node.block
+
+        arguments = node.arguments&.arguments
+        first_arg = arguments&.first
+        return unless first_arg.is_a?(Prism::StringNode)
+
+        test_name = first_arg.content
+        return if test_name.empty?
+
+        add_test_item(node, test_name)
+      end
+
+      #: (Prism::DefNode node) -> void
+      def on_def_node_enter(node)
+        return if @visibility_stack.last != :public
+
+        name = node.name.to_s
+        return unless name.start_with?("test_")
+
+        add_test_item(node, name)
+      end
+
+      private
+
+      #: (Array[String] attached_ancestors, String fully_qualified_name) -> bool
+      def declarative_minitest?(attached_ancestors, fully_qualified_name)
+        return false unless attached_ancestors.include?("Minitest::Test")
+
+        # Check if this is a regular Minitest test class that doesn't use ActiveSupport's
+        # declarative testing syntax. We need to handle both ActiveSupport::TestCase tests
+        # and plain Minitest tests.
```

Comment:
I just tested this on the Rails add-on codebase itself and test classes actually don't inherit from `Minitest::Test`. We need to remove this early return and then everything works (and the comment no longer applies either).

```suggestion
        # The declarative test style is present as long as the class extends
        # ActiveSupport::Testing::Declarative
```

Reply:
Thanks for checking, I'll remove the early return.

I had pushed an amendment to the comment (will update again) but it's still stuck "processing updates" today 🤔 

<details><summary>Img</summary>
<p>

![image](https://github.com/user-attachments/assets/add97591-7055-47f2-93cd-2e096ae469ab)


</p>
</details>